pkg/activator/certificate/cache.go

```diff
@@ -92,18 +134,24 @@ func (cr *CertCache) updateCache(secret *corev1.Secret) {
 	}
 	cr.certificate = &cert
 
-	pool := x509.NewCertPool()
 	block, _ := pem.Decode(secret.Data[certificates.CaCertName])
+	if block == nil {
+		cr.logger.Warnw("failed to parse CA")
+		return
+	}
 	ca, err := x509.ParseCertificate(block.Bytes)
 	if err != nil {
 		cr.logger.Warnw("failed to parse CA", zap.Error(err))
 		return
 	}
-	pool.AddCert(ca)
-
-	cr.TLSConf.RootCAs = pool
-	cr.TLSConf.ServerName = certificates.LegacyFakeDnsName
-	cr.TLSConf.MinVersion = tls.VersionTLS13
+	for _, usedCa := range cr.caCerts {
+		if usedCa.Equal(ca) {
+			return
+		}
+	}
+	cr.caCerts = append(cr.caCerts, ca)
+	cr.ClientTLSConf.RootCAs.AddCert(ca)
+	cr.ServerTLSConf.ClientCAs.AddCert(ca)
```

Comment:
`CertPool.AppendCertsFromPEM` will dedupe for you, parse multiple certs in the PEM block etc.